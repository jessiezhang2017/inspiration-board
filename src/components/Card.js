import React, { Component } from 'react';
import PropTypes from 'prop-types';
import emoji from 'emoji-dictionary';

import './Card.css';

class Card extends Component {

  render() {

    return (
      <div className="card">
        <section className="card__content">
          <p className="card__content-text">{this.props.text}</p>

          {this.props.emoji && (
           <p className="card__content-emoji">
             {emoji.getUnicode(this.props.emoji)
               ? emoji.getUnicode(this.props.emoji)
               : this.props.emoji}
           </p>
         )}
         <button
           onClick={() => this.props.deleteCardCallback(this.props.id)}
           type="button"
           className="card__content-delete-btn"
           aria-label="Close"
         >
         Delete
         </button>
        </section>
      </div>
    );
  }
}

Card.propTypes = {
   text: PropTypes.string.isRequired,
   emoji1: PropTypes.string,
}

export default Card;
