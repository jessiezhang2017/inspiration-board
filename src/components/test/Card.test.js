import React from 'react';
import Card from '../Card';
import {shallow} from 'enzyme';
// import App from './App';

describe('card',()=>{
  it('will match the Card Snapshot', ()=>{
    const wrapper = shallow(<Card
      text="go"
      emoji="heart_eyes"
      key={1}
      id={1}
      deleteCardCallback={()=>{}}

      />);
    expect(wrapper).toMatchSnapshot();
  });


});
