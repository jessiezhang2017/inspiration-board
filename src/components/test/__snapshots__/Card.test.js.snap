// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`card will match the Card Snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    emoji="heart_eyes"
    text="go"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <section
        className="card__content"
      >
        <p
          className="card__content-text"
        >
          go
        </p>
        <p
          className="card__content-emoji"
        >
          üòç
        </p>
      </section>,
      "className": "card",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p
            className="card__content-text"
          >
            go
          </p>,
          <p
            className="card__content-emoji"
          >
            üòç
          </p>,
        ],
        "className": "card__content",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "go",
            "className": "card__content-text",
          },
          "ref": null,
          "rendered": "go",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "üòç",
            "className": "card__content-emoji",
          },
          "ref": null,
          "rendered": "üòç",
          "type": "p",
        },
      ],
      "type": "section",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <section
          className="card__content"
        >
          <p
            className="card__content-text"
          >
            go
          </p>
          <p
            className="card__content-emoji"
          >
            üòç
          </p>
        </section>,
        "className": "card",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p
              className="card__content-text"
            >
              go
            </p>,
            <p
              className="card__content-emoji"
            >
              üòç
            </p>,
          ],
          "className": "card__content",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "go",
              "className": "card__content-text",
            },
            "ref": null,
            "rendered": "go",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "üòç",
              "className": "card__content-emoji",
            },
            "ref": null,
            "rendered": "üòç",
            "type": "p",
          },
        ],
        "type": "section",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
